scylla/tests/integration/session/schema_agreement.rs

```diff
@@ -156,3 +158,81 @@ async fn test_schema_await_with_unreachable_node() {
         Err(err) => panic!("{}", err),
     }
 }
+
+// Verifies that schema agreement process works correctly even if first check fails.
+#[tokio::test]
+#[cfg_attr(scylla_cloud_tests, ignore)]
+async fn test_schema_await_with_transient_failure() {
+    setup_tracing();
+
+    let res = test_with_3_node_cluster(
+        ShardAwareness::QueryNode,
+        |proxy_uris, translation_map, mut running_proxy| async move {
+            // DB preparation phase
+            let session: Session = SessionBuilder::new()
+                .known_node(proxy_uris[0].as_str())
+                .address_translator(Arc::new(translation_map.clone()))
+                // Important in order to have a predicatable amount of connections after session creation.
+                // Shard connections are created asynchronously, so its hard to predict how many will be opened
+                // already when we check schema agreement.
+                .pool_size(PoolSize::PerHost(1.try_into().unwrap()))
+                .build()
+                .await
+                .unwrap();
+
+            let node_rules = Some(vec![RequestRule(
+                Condition::not(Condition::ConnectionRegisteredAnyEvent)
+                    .and(Condition::RequestOpcode(RequestOpcode::Query))
+                    .and(Condition::BodyContainsCaseSensitive(Box::new(
+                        *b"system.local",
+                    )))
+                    .and(Condition::TrueForLimitedTimes(1)),
+                // Use error that would prevent DefaultRetryPolicy from retrying.
+                // I don't think it is used for those queries, but its additional future-proofing
+                // for the test.
+                RequestReaction::forge_with_error(DbError::SyntaxError),
```

Comment:
:thinking: Well... this definitely isn't a transient error, and it definitely should result in breaking out of the schema agreement awaiting loop.